lib/recurly/bus.js

```diff
@@ -35,29 +38,12 @@ export class Bus extends Emitter {
    * @private
    * @param {MessageEvent} message
    */
-  receive (message) {
-    if (!this.originMatches(message.origin)) return;
-    if (typeof message.data === 'string') return this.receiveFromRelay(message)
-    if (!message.data.event) return;
-    this.debug('message received', message.data);
-    this.send(new Message(message), null, { exclude: [message.srcElement] });
-  }
-
-  /**
-   * Parses message properties from a JSON-encoded string
-   *
-   * @private
-   * @param  {String} message
-   * @return {Object} message
-   */
-  receiveFromRelay (message) {
-    let data;
-    try {
-      data = JSON.parse(message.data);
-    } catch (e) {
-      this.debug('failed to parse a string message', e, message);
-    }
-    if (data) this.send(new Message(data.recurly_event, data.recurly_message));
+  receive (messageEvent) {
+    if (!this.originMatches(messageEvent.origin)) return;
+    this.debug('message event received', messageEvent.data);
+    const message = Message.createFromMessageEvent(messageEvent, this.debug.bind(this));
```

Comment:
There was too much replication of concerns previously with the `receiveFromRelay` method, somewhat a holdover from v3. This new method handles object translation and error handling for normal messages and those sent via a relay,